_posts/2020-06-12-terraform-azurerm-azure-container-instance-devops-agent.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: post
+title:  "Run Azure DevOps self-hosted agents in Azure Container Instance using Terraform"
+date:   2020-05-12 10:00:00 +0200
+categories: 
+- Microsoft Azure
+- Terraform
+- Azure DevOps
+author: 'Julien Corioland'
+identifier: 'd33c206c-f679-4dcf-ad5b-f235193a633b'
+image: /images/terraform-microsoft-azure-introduction/terraform-azure.png
+---
+
+Azure DevOps offers two kinds of pipeline agents: Microsoft-hosted agents, fully managed by Microsoft, and self-hosted agents, fully managed by yourself. I recently worked with a customer who needed to be able to run the pipeline agents into their private virtual network infrastructure which is one of the use case for having to use self-hosted agents.
+
+<!--more-->
+
+Because a lot of customers are asking for being able to spin up quickly Azure DevOps self-hosted agents, I have decided to develop a [Terraform module](https://github.com/Azure/terraform-azurerm-aci-devops-agent) that can help anyone to deploy one (or more) into a Docker container and host it quickly into [Azure Container Instance](https://azure.microsoft.com/en-us/services/container-instances/). By doing that you can have one ore more agents up and running in only a few minutes. Very useful when you have build workloads that require to scale up & down very frequently and that you want to minimize costs of hosting build agents.
```

Comment:
```suggestion
Because a lot of customers are asking for being able to spin up quickly Azure DevOps self-hosted agents, I have decided to develop a [Terraform module](https://github.com/Azure/terraform-azurerm-aci-devops-agent) that can help anyone to deploy one (or more) self-hosted agents as Docker containers and host it quickly on [Azure Container Instances](https://azure.microsoft.com/en-us/services/container-instances/). Followin that approach you can have one ore more agents up and running in only a few minutes. Very useful when you have workloads that require to scale up & down very frequently and you want to minimize the cost of hosting.
```